tools/examples/components/sql-scratchpad/src/components/SqlScratchpad.tsx

```diff
@@ -1,85 +1,81 @@
-import React from 'react';
+import React, { FC, useEffect, useState } from 'react';
 
 import hueComponents from 'gethue/lib/components/QueryEditorWebComponents';
-import hueConfig from 'gethue/lib/config/hueConfig';
 import Executor from 'gethue/lib/execution/executor';
 import SqlExecutable from 'gethue/apps/editor/execution/sqlExecutable';
 
 import { QueryEditor } from './QueryEditor';
 import { ExecuteButton } from './ExecuteButton';
 import { ExecuteProgress } from './ExecuteProgress';
 import { ResultTable } from './ResultTable';
-import {ExecuteLimit} from "./ExecuteLimit";
+import { ExecuteLimit } from "./ExecuteLimit";
+
+const { configure, refreshConfig, findEditorConnector, createExecutor, getNamespaces } = hueComponents;
 
 const HUE_BASE_URL = 'http://localhost:8888'
 
-hueComponents.configure({
+configure({
   baseUrl: HUE_BASE_URL
 });
 
-interface SqlScratchpadState {
-  activeExecutable?: SqlExecutable;
-  executor?: Executor;
-}
-
-type KnockoutObservable<T> = () => T;
+export const SqlScratchpad: FC = () => {
+  const [activeExecutable, setActiveExecutable] = useState<SqlExecutable | undefined>(undefined);
+  const [executor, setExecutor] = useState<Executor | undefined>(undefined);
 
-export class SqlScratchpad extends React.Component<{}, SqlScratchpadState> {
-  state = {
-    activeExecutable: undefined,
-    executor: undefined
-  }
+  const setup = async () => {
```

Comment:
And it is getting a pain to rebase in compose-ui, so will be a good scenario/usage test to see how to make it clean in the new SS!

e.g. integrate auth back, easy to override etc
https://github.com/gethue/compose/commit/17b0c4111f404ef64fd0117dc39887a9cce98740#diff-0e8ed77e970aa4715ffe66619a107e3ce1ffae6c5f47f9f9f81234e1b406d2acR47